### PR TITLE
Add KafkaSink upgrade tests

### PR DESCRIPTION
**Pull request by pierDipi**
To implement KafkaSink upgrade tests we reuse an existing utility
that runs a KafkaSink test provided a `KafkaSinkSpec` object.

The continual test for KafkaSink is implemented using a combination
`KafkaSink` and `KafkaSource` as follows:

```
-> Upgrade test source
  -> `KafkaSink`
    -> `KafkaSource`
      -> Upgrade test sink
```

We use KafkaSink and KafkaSource to avoid creating a specific
kafka consumer for the upgrade test.

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

**Review by mgencur — COMMENTED** [review]
Looks good! Just one minor nit about logging. But feel free to ignore it.
There's now a little overlap between the `continual.SourceTest` in https://github.com/knative-sandbox/eventing-kafka and this new continual test which uses both Kafka Source and Kafka Sink. But I think that's alright.

File: test/e2e/kafka_sink.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			NumPartitions:     pointer.Int32Ptr(10),
+			ReplicationFactor: func(rf int16) *int16 { return &rf }(1),
+			BootstrapServers:  BootstrapServersPlaintextArr,
+			ContentMode:       pointer.StringPtr(mode),
+		}
+		for _, opt := range opts {
+			require.Nil(t, opt(&kss))
+		}
+
+		t.Log(kss)
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/e2e/kafka_sink.go:70]: +1, thanks!

**Comment by cardil** [comment]
Looks good to me, too. Nice work!

File: test/e2e/kafka_sink.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/e2e/kafka_sink.go]: Done

**Review by cardil — APPROVED** [review]
/lgtm

**Pull request merged**